### PR TITLE
Refactor documentation

### PR DESCRIPTION
**Pull request by huwb**
This does a major pass on documentation. Moves structure from system-centric (like grouping all loddata types together) to user-centric (grouping things by category, focusing around things users want to do).

This has the high level structure in place and is ready for a high level review i'd say (but detailed notes welcome if you have them).

Docs build: https://crest.readthedocs.io/en/refactor-docs/user/quick-start-guide.html

### CODE REVIEW

**Comment by daleeidd** [comment]
Just FYI there is no requirement to rename files to match the heading. RTDs will use the heading name.

**Comment by daleeidd** [comment]
If you rename files you'll likely need to go through the C# code and update links there too.

**Comment by huwb** [comment]
Thanks yeah there are a couple that differ between file and section header. The ones I changed were more to keep it tidy. But if you want some of them dropped lmk. If not i'll update the C# - good shout!

**Comment by huwb** [comment]
Pheww think URLs updated although will need to do another round of testing before merge

File: docs/user/water-appearance.rst [hunk 1]
```diff
@@ -381,6 +349,182 @@ Anything rendered in the transparent pass or higher will not be included in refr
...
+Crest detects where waves are 'pinched' and deposits foam to approximate whitecaps.
+
+-  **Wave Foam Strength** - Scales intensity of foam generated from waves.
+   This setting should be balanced with the *Foam Fade Rate* setting.
+
+-  **Wave Foam Coverage** - How much of the waves generate foam.
+   Higher values will lower the threshold for foam generation, giving a larger area.
+
+
+Shoreline foam
```
Comment (daleeidd) [inline docs/user/water-appearance.rst:420]: This could be referenced in shorelines page.

**Review by daleeidd — COMMENTED** [review]
This is a top-level review. An overall improvement. Great work.

Looking at the new structure it is good to see the _Ocean Simulation_ page gone.

I think _Authoring Input Modes_ could be improved too. It could be named something like "Editing Water" or "Water Inputs" which will draw users in and should contain an overview of inputs in general  which I think is now missing. _Basics of Adding Ocean Inputs_ video covered it even if outdated which could be added back in and replaced in the future.

File: docs/user/water-bodies.rst [hunk 2]
```diff
@@ -49,3 +49,119 @@ The *Add Feature* section of the *Spline* inspector has helper buttons to quickl
...
+The *Crest/Inputs/Flow/Add Flow Map* shader writes a flow texture into the system.
+It assumes the x component of the flow velocity is packed into 0-1 range in the red channel, and the z component of the velocity is packed into 0-1 range in the green channel.
+The shader reads the values, subtracts 0.5, and multiplies them by the provided scale value on the shader.
+The process of adding ocean inputs is demonstrated in :numref:`adding-inputs-video`.
+
+
+
+.. _clip-surface-section:
+
+Clip Surface
```
[the remark this answers is not in the document]
Comment (huwb) [reply docs/user/water-bodies.rst:94]: yeah.. i think thats a good point. i did think about it a bit.. i wasnt actually sure what to call it. I'm not sure *clipping* means anything to the average user. but i cant think of anything better. can you? if not i'll create a clipping page.

[the remark this answers is not in the document]
Comment (daleeidd) [reply docs/user/water-bodies.rst:94]: Unity has _Terrain Holes_ so maybe _Holes_? Otherwise _Masking_ or _Placement_ (a bit more adventurous)...

**Pull request merged**